content/hoon/lagoon/build.md

```diff
@@ -0,0 +1,362 @@
+## `+eye` {#eye}
+
+Identity matrix of shape nxn.
+
+#### Accepts
+
+A `$meta`
+
+#### Produces
+
+A `$ray`
+
+#### Source
+
+```hoon
+++  eye      ::  produces identity matrix of shape nxn.
+  |=  =meta
+  ^-  ray
+  ~_  leaf+"lagoon-fail"
+  ?>  =(2 (lent shape.meta))
+  ?>  =((snag 0 shape.meta) (snag 1 shape.meta))
+  =/  n  (snag 0 shape.meta)
+  =<  +
+  %^    spin
+      (gulf 0 (dec n))
+    ^-  ray  (zeros [~[n n] bloq.meta kind.meta ~])
+  |=  [i=@ r=ray]
+  :: [i (set-item r ~[i i] 1)]
+  :-  i
+  %^  set-item
+      r
+    ~[i i]
+  ^-  @
+  ?-    kind.meta
+      %uint  `@`1
+    ::
+      %int2  !!
+    ::
+      %i754
+    ?+  bloq.meta  ~|(bloq.meta !!)
+      %7  .~~~1
+      %6  .~1
+      %5  .1
+      %4  .~~1
+    ==
+  ==
+```
+
+---
+
+## `+zeros` {#zeros}
+
+Zeroes
```

Comment:
Throughout the docs there are these arm descriptions which have a redundant paragraph like this at the beginning, we could live without them. I think this only occurs in this arm and the few below it. Should either be expanded to a more full description of the why/how/what of it all or omitted entirely.